### PR TITLE
Users/tasoni/port c chub changes

### PR DESCRIPTION
**Pull request by tanvi-soni**
porting changes from https://mseng.visualstudio.com/VSOnline/Test%20Infrastructure/_git/VSO/pullrequest/93732?view=discussion

### CODE REVIEW

File: src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs [hunk 1]
```diff
@@ -87,6 +88,43 @@ public void PublishCodeCoverageWithInvalidCCTool()
...
+                var command = new Command("codecoverage", "publish");
+                command.Properties.Add("codecoveragetool", "cobertura");
+                command.Properties.Add("summaryfile", coberturaXml);
+                command.Properties.Add("reportdirectory", reportDirectory);
+                publishCCCommand.ProcessCommand(_ec.Object, command);
+                Assert.Equal(0, _warnings.Count);
+                Assert.Equal(0, _errors.Count);
+                _mockCodeCoveragePublisher.Verify(x => x.PublishCodeCoverageSummaryAsync(It.IsAny<IEnumerable<CodeCoverageStatistics>>(), It.IsAny<string>(), It.IsAny<CancellationToken>()));
+                _mockCodeCoveragePublisher.Verify(x => x.PublishCodeCoverageFilesAsync(It.IsAny<IAsyncCommandContext>(), It.IsAny<Guid>(), It.IsAny<long>(),
+                  It.IsAny<List<Tuple<string, string>>>(), It.Is<bool>(browsable => browsable == true), It.IsAny<CancellationToken>()));
```
Comment (prawalagarwal) [inline src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs]: spacing

File: src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs [hunk 2]
```diff
@@ -270,6 +272,27 @@ private void LoadPublishCodeCoverageInputs(Dictionary<string, string> eventPrope
...
+        /// </summary>
+        /// <param name="reportDirectory"></param>
+        private void ModifyCoberturaIndexDotHTML(string reportDirectory, IExecutionContext executionContext)
+        {
+            try
+            {
+                if (_codeCoverageTool.Equals("cobertura", StringComparison.OrdinalIgnoreCase) && File.Exists(Path.Combine(reportDirectory, CodeCoverageConstants.DefaultIndexFile)) && File.Exists(Path.Combine(reportDirectory, CodeCoverageConstants.DefaultNonFrameFileCobertura)))
+                {
+                    File.Move(Path.Combine(reportDirectory, CodeCoverageConstants.DefaultIndexFile), Path.Combine(reportDirectory, CodeCoverageConstants.NewIndexFile));
+                    File.Move(Path.Combine(reportDirectory, CodeCoverageConstants.DefaultNonFrameFileCobertura), Path.Combine(reportDirectory, CodeCoverageConstants.DefaultIndexFile));
```
Comment (prawalagarwal) [inline src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs]: You have not considered changes in [this](https://mseng.visualstudio.com/VSOnline/Test%20Infrastructure/_git/VSO/pullrequest/110815#view=discussion&fullScreen=false) and [this](https://mseng.visualstudio.com/VSOnline/Test%20Infrastructure/_git/VSO/pullrequest/112413?view=discussion) PR. Please incorporate these changes.

File: src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs (hunk 1 above)
Comment (tanvi-soni) [inline src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs]: done

File: src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs [hunk 3]
```diff
@@ -87,6 +88,42 @@ public void PublishCodeCoverageWithInvalidCCTool()
 
         [Fact]
         [Trait("Level", "L0")]
+        public void Publish_CoberturaNewIndexFile()
```
Comment (prawalagarwal) [inline src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs]: Ran the tests locally?

File: src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs [hunk 4]
```diff
@@ -172,6 +172,8 @@ private void ProcessPublishCodeCoverageCommand(IExecutionContext context, Dictio
                 Directory.CreateDirectory(Path.GetDirectoryName(destinationSummaryFile));
                 File.Copy(_summaryFileLocation, destinationSummaryFile, true);
 
+                commandContext.Output("Modifying Cobertura Index file");
```
Comment (TingluoHuang) [inline src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs]: LOC

File: src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs [hunk 5]
```diff
@@ -270,6 +272,28 @@ private void LoadPublishCodeCoverageInputs(Dictionary<string, string> eventPrope
...
+        /// <summary>
+        /// This method replaces the default index.html generated by cobertura with 
+        /// the non-framed version
+        /// </summary>
+        /// <param name="reportDirectory"></param>
+        private void ModifyCoberturaIndexDotHTML(string reportDirectory, IExecutionContext executionContext)
+        {
+            try
+            {
+                if (_codeCoverageTool.Equals("cobertura", StringComparison.OrdinalIgnoreCase) && File.Exists(Path.Combine(reportDirectory, CodeCoverageConstants.DefaultIndexFile)) && File.Exists(Path.Combine(reportDirectory, CodeCoverageConstants.DefaultNonFrameFileCobertura)))
```
Comment (acesiddhu) [inline src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs]: suggest to use string.Equals(string1, string2, StringComparison.OrdinalIgnoreCase) as a practice

File: src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs [hunk 6]
```diff
@@ -87,6 +88,42 @@ public void PublishCodeCoverageWithInvalidCCTool()
...
+        {
+            SetupMocks();
+            var reportDirectory = Path.Combine(Path.GetTempPath(), "reportDirectory");
+            var coberturaXml = Path.Combine(reportDirectory, "coberturaValid.xml");
+
+            try
+            {
+                Directory.CreateDirectory(reportDirectory);
```
Comment (acesiddhu) [inline src/Test/L0/Worker/CodeCoverage/CodeCoverageCommandExtensionTests.cs]: L0 tests shuldnt be doing this. Correct?

**Pull request merged**